pontoon/messaging/static/js/messaging.js

```diff
@@ -162,13 +163,87 @@ $(function () {
     $('#review .message-type .transactional').toggle(isTransactional);
   }
 
+  function loadPanelContent(path) {
+    if (inProgress) {
+      inProgress.abort();
+    }
+
+    const panel = container.find('.right-column');
+    const relative_path = path.split('/messaging/')[1];
+    const menu_path = path === '/messaging/sent/' ? path : '/messaging/';
+
+    // Update menu selected state
+    container
+      .find(`.left-column a[href="${menu_path}"]`)
+      .parents('li')
+      .addClass('selected')
+      .siblings()
+      .removeClass('selected');
+
+    panel.empty();
+
+    inProgress = $.ajax({
+      url: `/messaging/ajax/${relative_path}`,
+      success: function (data) {
+        panel.append(data);
+
+        if (relative_path === 'sent/') {
+          // Dissolve new message state
+          setTimeout(function () {
+            container.find('.message.new').removeClass('new');
+          }, 1000);
+        } else if (relative_path) {
+          // Convert body content from HTML to markdown
+          const html = $('#compose [name=body]').val();
+          const markdown = converter.makeMarkdown(html);
+          $('#body').val(markdown);
+        }
+      },
+      error: function (error) {
+        if (error.status === 0 && error.statusText !== 'abort') {
+          const message = $('<p>', {
+            class: 'no-results',
+            html: 'Oops, something went wrong.',
+          });
+
+          panel.append(message);
+        }
+      },
+    });
+  }
+
+  // Load panel conent on page load
+  loadPanelContent(window.location.pathname);
+
+  // Load panel conent on history change
+  window.onpopstate = function () {
+    loadPanelContent(window.location.pathname);
+  };
+
+  // Load panel conent on menu click
+  container.on(
+    'click',
+    '.left-column a, .right-column .edit-as-new',
+    function (e) {
+      // Keep default middle-, control- and command-click behaviour (open in new tab)
+      if (e.which === 2 || e.metaKey || e.ctrlKey) {
+        return;
+      }
```

Comment:
How about shift+click?

Reply:
We'd need to file a bug for that and change it across all pages with tabs.

Reply:
The translate frontend already checks for `shiftKey` when it checks for `ctrlKey`, so we shouldn't deviate from that more in the places where we do special stuff on clicks.
```suggestion
      // Keep default middle-, control- and command-click behaviour (open in new tab)
      if (e.which === 2 || e.metaKey || e.ctrlKey || e.shiftKey) {
        return;
      }
```

Reply:
I meant this: https://github.com/mozilla/pontoon/blob/main/pontoon/base/static/js/tabs.js#L22

Reply:
I'll update both files.